src/ctap/hid/mod.rs

```diff
@@ -51,79 +88,84 @@ pub enum ProcessedPacket<'a> {
     },
 }
 
-// An assembled CTAPHID command.
+/// An assembled CTAPHID command.
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Message {
     // Channel ID.
     pub cid: ChannelID,
     // Command.
-    pub cmd: u8,
+    pub cmd: CtapHidCommand,
     // Bytes of the message.
     pub payload: Vec<u8>,
 }
 
+/// A keepalive packet reports the reason why a command does not finish.
+#[allow(dead_code)]
+pub enum KeepaliveStatus {
+    Processing = 0x01,
+    UpNeeded = 0x02,
+}
+
+/// Holds all state for receiving and sending HID packets.
+///
+/// This includes
+/// - state from not fully processed messages,
+/// - all allocated channels,
+/// - information about requested winks.
+///
+/// The wink information can be polled to decide to i.e. blink LEDs.
+///
+/// To process a packet and receive the response, you can call `process_hid_packet`.
+/// If you want more control, you can also do the processing in steps:
+///
+/// 1.  `HidPacket` -> `Option<Message>`
+///     `parse_packet` assembles the message and preprocesses all pure HID commands and errors.
+/// 2.  `Option<Message>` -> `Message`
+///     If you didn't receive any message or preprocessing discarded it, stop.
+/// 3.  `Message` -> `Message`
+///     `process_message` handles all protocol interactions.
+/// 4.  `Message` -> `HidPacketIterator
```

Comment:
Missing backtick. Did you try `cargo doc --document-private-items --features=std --open` to check the output? (there's also wrong indentation line 124)

Reply:
Fixed. Thanks for the pointer. I noticed the formatting didn't work the way I wanted it, so I changed it a bit. Also added more documentation for what I think is the forseeable public interface.